app/uk/gov/hmrc/exports/services/NotificationService.scala

```diff
@@ -84,21 +92,26 @@ class NotificationService @Inject()(submissionRepository: SubmissionRepository,
     }
 
   private def updateRelatedSubmission(notification: Notification): Future[Either[String, Unit]] =
-    try {
-      submissionRepository.updateMrn(notification.actionId, notification.mrn).map {
-        case None =>
-          logger.error(s"Could not find Submission to update for actionId: ${notification.actionId}")
-          Right((): Unit)
-        case Some(_) =>
-          Right((): Unit)
+    notification.details.map { details =>
+      try {
+        submissionRepository.updateMrn(notification.actionId, details.mrn).map {
+          case None =>
+            logger.error(s"Could not find Submission to update for actionId: ${notification.actionId}")
+            Right((): Unit)
+          case Some(_) =>
+            Right((): Unit)
+        }
+      } catch {
+        case exc: Throwable =>
+          logger.error(exc.getMessage)
+          Future.successful(Left(exc.getMessage))
       }
-    } catch {
-      case exc: Throwable =>
-        logger.error(exc.getMessage)
-        Future.successful(Left(exc.getMessage))
-    }
+    }.getOrElse(Future.successful(Right((): Unit)))
 
   def buildNotificationsFromRequest(notificationApiRequestHeaders: NotificationApiRequestHeaders, notificationXml: NodeSeq): Seq[Notification] =
```

Comment:
I'm wondering if this responsibility should not be extracted into separate builder or converter class.